x-pack/plugins/observability/public/components/shared/alert_status_indicator/alert_status_indicator.test.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { shallow } from 'enzyme';
```

Comment:
Regarding using `enzyme`, I checked and it seems we don't have a common approach.

Some components in the `observability` plugin seem to be using [this helper function](https://github.com/afgomez/kibana/blob/f36ae806382a994d4bcb644a413d5e8b28c5b31e/x-pack/plugins/observability/public/utils/test_helper.tsx#L45-L64), which sets up some stuff that might help with the issues you had with `useUiSetting`. Give it a try but if it doesn't work don't stress too much about it.

Reply:
I have no idea if it does shallow rendering 🤷

Reply:
Following up on this:
I've now re-written the tests to use `@testing-library/react` (TLR), I found this article helpful https://medium.com/wesionary-team/react-testing-library-vs-enzyme-afd29db380ac in understanding the choice between TLR and Enzyme. 

TLR does not do shallow rendering since it's goal is for you to make assertions on the resulting DOM to avoid testing implementation details.